components/usage/pkg/stripe/stripe.go

```diff
@@ -66,36 +71,57 @@ func (c *Client) UpdateUsage(ctx context.Context, creditsPerTeam map[string]int6
 		iter := c.sc.Customers.Search(params)
 		for iter.Next() {
 			customer := iter.Customer()
-			log.Infof("found customer %q for teamId %q", customer.Name, customer.Metadata["teamId"])
-			subscriptions := customer.Subscriptions.Data
-			if len(subscriptions) != 1 {
-				log.Errorf("customer has an unexpected number of subscriptions (expected 1, got %d)", len(subscriptions))
-				continue
-			}
-			subscription := customer.Subscriptions.Data[0]
-
-			log.Infof("customer has subscription: %q", subscription.ID)
-			if len(subscription.Items.Data) != 1 {
-				log.Errorf("this subscription has an unexpected number of subscriptionItems (expected 1, got %d)", len(subscription.Items.Data))
-				continue
-			}
-
-			creditsUsed := creditsPerTeam[customer.Metadata["teamId"]]
+			teamID := customer.Metadata["teamId"]
+			log.Infof("Found customer %q for teamId %q", customer.Name, teamID)
 
-			subscriptionItemId := subscription.Items.Data[0].ID
-			log.Infof("registering usage against subscriptionItem %q", subscriptionItemId)
-			_, err := c.sc.UsageRecords.New(&stripe.UsageRecordParams{
-				SubscriptionItem: stripe.String(subscriptionItemId),
-				Quantity:         stripe.Int64(creditsUsed),
-			})
+			_, err := c.updateUsageForCustomer(ctx, customer, creditsPerTeam[teamID])
 			if err != nil {
-				log.WithError(err).Errorf("failed to register usage for customer %q", customer.Name)
+				log.WithField("customer_id", customer.ID).
+					WithField("customer_name", customer.Name).
+					WithField("subscriptions", customer.Subscriptions).
+					WithError(err).
+					Errorf("Failed to update usage.")
+
+				reportStripeUsageUpdate(err)
+				continue
 			}
+			reportStripeUsageUpdate(nil)
 		}
 	}
 	return nil
 }
 
+func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Customer, credits int64) (*UsageRecord, error) {
```

Comment:
The `*UsageRecord` return value is ignored by its only caller. is that intentional? If so can we remove the return value here?

Reply:
I was planning to add a second metric which shows total amount of usage increase as well as return it in the RPC. You're right, it's not needed in this PR directly. I can remove it if you want

Reply:
I'll leave it up to you. No point in removing if the next PR puts it straight back.